polyaxon_cli/cli/run.py

```diff
@@ -135,6 +137,9 @@ def run(ctx,
     tags = validate_tags(tags)
 
     if local:
+        if ctx.obj["offline"]:
+            os.environ['POLYAXON_IS_OFFLINE'] = 'true'
+            settings.IS_OFFLINE = True
```

Comment:
Is this required?

Reply:
Polyaxon-client uses `settings.IS_OFFLINE` to check for offline mode.

settings monkeypatching is not required if we import `polyaxon_client.settings` after `POLYAXON_IS_OFFLINE` env variable is set.

Reply:
I understand, but I don't see why this change here, I would either do that on the main close to where we set the offline on the context, this way any command check the `offline` flag does not have to duplicate code, or not do it at all, as far as I can see, we only need to check for the offline, at the moment, in the local mode to add an export to the dockerfile, which we can do just with the env vars.

Reply:
In local mode we need to create BuildJob and Experiment instances - they also need offline mode.

Reply:
I moved offline mode handling to `polyaxon_cli/main.py`